src/withForm.ts

```diff
@@ -0,0 +1,192 @@
+  class LiteForm extends Component {
+    values!: ValueMap;
+    errors!: ErrorMap;
+    touched!: TouchedMap;
+    isValid!: boolean | undefined; 
+
+    _onSubmit!: (values: ValueMap, props: FormElementProps) => void;
+    _initialValues!: ValueMap;
+    _validationSchema!: ValidatorMap;
+    _validateOnBlur!: boolean;
+    _validateOnChange!: boolean;
+
+    connectedCallback(): void {
+      const {onSubmit, initialValues, validationSchema, validateOnChange, validateOnBlur} = config ?? {};
+      // take params from HOC argument, or from class (if you build base class, like <lite-form>) or default
+      this._onSubmit = (onSubmit || this.onSubmit || function () {}).bind(this)
+      this._initialValues = initialValues || this.initialValues || {}
+      this._validationSchema = validationSchema || this.validationSchema || {}
+      this._validateOnBlur = 
+        validateOnBlur ??
+        this.validateOnBlur ??
+        true
+      this._validateOnChange = validateOnChange ??
+        this.validateOnChange ??
+        true
+
+      // isLiteForm atribute for HOCs can find their form element
+      this.setAttribute(IS_LITE_FORM, 'true')
+
+      this.handleReset()
+      super.connectedCallback && super.connectedCallback()
+    }
+
+    get _values(): ValueMap {
+      return this.values
+    }
+
+    set _values(values: ValueMap) {
+      this.values = values
+
+      this.dispatchEvent(
+        new CustomEvent(EVENTS.valuesChange, {
+          detail: { values: this.values }
+        })
+      )
+    }
+
+    get _errors(): ErrorMap {
+      return this.errors
+    }
+
+    set _errors(errors: ErrorMap) {
+      this.errors = errors
+
+      this.dispatchEvent(
+        new CustomEvent(EVENTS.errorsChange, {
+          detail: { errors: this.errors, touched: this.touched }
+        })
+      )
+    }
+
+    get _touched(): TouchedMap {
+      return this.touched
+    }
+
+    set _touched(touched: TouchedMap) {
+      this.touched = touched
+      this.dispatchEvent(
+        new CustomEvent(EVENTS.errorsChange, {
+          detail: { errors: this.errors, touched: this.touched }
+        })
+      )
+    }
+
+    handleReset = (e?:Event): void => {
+      e && e.preventDefault()
+      this._values = cloneDeep(this._initialValues)
+      this._touched = {}
+      this._errors = {}
+      this.isValid = undefined
+    }
+
+    handleSubmit = (e?:Event): void => {
+      e && e.preventDefault()
+
+      this.handleValidate()
+
+      if (this.isValid) {
+        this._onSubmit(this._values, this)
+      } else {
+        // Touch all fields to show validation errors
+        this._touched = Object.keys(this._validationSchema).reduce(
+          (obj: { [propName: string]: boolean; }, fieldName: string) => {
+            set(obj, fieldName, true)
+            return obj
+          },
+          {}
+        )
+      }
+    }
+
+    // Set touched and handleValidate if needed
+    setTouched = (name: string): void => {
```

Comment:
Do we need another argument 'validate = false' here, for user can use it without validation. 
Also call it with  setTouched(name, true) from withForm. handleBlur()

Reply:
and 
if (this._validateOnBlur) {
  this.handleValidate()
}   
`
can be a part of withForm.handleBlur() - think it will be more expected behavior

Reply:
Done.